src/common/nsat.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+
+import { commands, env, ExtensionContext, extensions, Uri, window } from "vscode";
+import { Constants } from "./constants";
+import { TelemetryClient } from "./telemetryClient";
+
+const NSAT_SURVEY_URL = "https://www.surveymonkey.com/r/JT2MQK5";
+const PROBABILITY = 0.5;
+const SESSION_COUNT_THRESHOLD = 2;
+const SESSION_COUNT_KEY = "nsat/sessionCount";
+const LAST_SESSION_DATE_KEY = "nsat/lastSessionDate";
+const TAKE_SURVEY_DATE_KEY = "nsat/takeSurveyDate";
+const DONT_SHOW_DATE_KEY = "nsat/dontShowDate";
+const SKIP_VERSION_KEY = "nsat/skipVersion";
+const IS_CANDIDATE_KEY = "nsat/isCandidate";
+
+export class NSAT {
+    public static async takeSurvey({ globalState }: ExtensionContext) {
+        const skipVersion = globalState.get(SKIP_VERSION_KEY, "");
```

Comment:
is this key maintained for later per version survey?

Reply:
Yes, in case we have fewer users to take survey